src/ILVerification/src/TypeVerifier.cs

```diff
@@ -98,30 +99,47 @@ public void VerifyInterfaces()
                         MethodDesc resolvedMethod = type.ResolveInterfaceMethodTarget(method);
                         if (resolvedMethod is null)
                         {
-                            VerificationError(VerifierError.InterfaceMethodNotImplemented, Format(type), Format(implementedInterface.DefType), Format(method));
+                            VerificationError(VerifierError.InterfaceMethodNotImplemented, Format(type), Format(implementedInterface.DefType, implementedInterface.InterfaceImplementationHandle), Format(method));
                         }
                     }
                 }
             }
         }
 
-        // Format helpers for future use
-        // The idea is to improve formatting string with tokens value i.e.: 
-        // [Assembly]Class(0x1234) vs current [Assembly]Class
-        // in case somebody will pass '--tokens' switch to ILVerify
         private string Format(EcmaType type)
         {
-            return type.ToString();
+            if (_verifierOptions.IncludeMetadataTokensInErrorMessages)
+            {
+                return $"{type}(0x{_module.MetadataReader.GetToken(type.Handle)})";
+            }
+            else
+            { 
+                return type.ToString();
+            }
         }
 
-        private string Format(DefType interfaceTypeDef)
+        private string Format(DefType defType, InterfaceImplementationHandle interfaceImplementationHandle)
         {
-            return interfaceTypeDef.ToString();
+            if (_verifierOptions.IncludeMetadataTokensInErrorMessages)
+            {
+                return $"{defType}(0x{_module.MetadataReader.GetToken(interfaceImplementationHandle)})";
+            }
+            else
+            {
+                return defType.ToString();
+            }
         }
 
         private string Format(MethodDesc methodDesc)
         {
-            return methodDesc.ToString();
+            if (_verifierOptions.IncludeMetadataTokensInErrorMessages)
+            {
+                return $"{methodDesc}(0x{_module.MetadataReader.GetToken(methodDesc.IsTypicalMethodDefinition ? ((EcmaMethod)methodDesc).Handle : ((EcmaMethod)methodDesc.GetTypicalMethodDefinition()).Handle)})";
```

Comment:
You can just call `GetTypicalMethodDefinition` without checking `IsTypicalMethodDefinition` first. `GetTypicalMethodDefinition` returns `this` if you call it on the typical method definition.

Reply:
I noticed...it seemed more clear code intention...but if it's ok to you I agree!